core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop;
+	int len_prop;
+	int it_num;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if ((int_prop == NULL) || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
+		const char *compatible)
+{
+	paddr_t offset = 0;
+	int fnode;
+	int node_status;
+
+	fnode = fdt_node_offset_by_compatible(fdt, *node,
+			compatible);
+
+	while (fnode != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, fnode);
+		if ((node_status & status) == status) {
+			offset = _fdt_reg_base_address(fdt, fnode);
+			*node  = fnode;
+			break;
+		}
+
+		fnode = fdt_node_offset_by_compatible(fdt, fnode,
+				compatible);
+	}
+
+	return offset;
+}
+
+int dt_disable_status(void *fdt, int node)
+{
+	const char *prop;
+	int len;
+
+	prop = fdt_getprop(fdt, node, "status", &len);
+	if (!prop) {
+		/* status is not available, just add it */
+		if (fdt_setprop_string(fdt, node, "status", "disabled"))
+			return -1;
+	} else {
+		/* status is there, modify it */
+		/* Actually we do not have to put disabled, anything that is
+		 * not "ok" or "okay" will work
+		 */
+		if (fdt_setprop_inplace(fdt, node, "status",
+				"disa", len))
+			return -1;
+	}
+
+	return 0;
+}
+
+int dt_set_secure_status(void *fdt, int node)
+{
+	const char *prop;
+	int len;
+
+	if (dt_disable_status(fdt, node)) {
+		EMSG("Unable to disable Normal Status");
+		return (-1);
+	}
+
+	prop = fdt_getprop(fdt, node, "secure-status", &len);
+	if (!prop) {
+		/* status is not available, just add it */
+		if (fdt_setprop_string(fdt, node, "secure-status", "okay"))
+			return -1;
+	} else {
+		/* status is there, modify it to okay */
+		if (fdt_setprop_inplace(fdt, node, "secure-status",
+				"okay", len))
```

Comment:
What if `len` is larger than `5`?
`secure-status` must be set to `ok` or `okay`, what if `len` is `< 2` or `3`?

Reply:
I'll change the function to use the `fdt_appendprop_string` that is taking care of either creating the entry or modifying it by setting the "okay" value and updating property length too.